File: Python Exams/Basic Exam Collection/Basics Online Exam - 20 and 21 April 2019/06. Easter Decoration/06. Easter Decoration.py
number_clients = int(input())
loop = number_clients
price_dict = { 'basket': 1.50, 'wreath' : 3.80, 'chocolate bunny' : 7}
total_sum = 0

while loop:
    loop-= 1
    current_sum = 0
    items_count = 0

    while True:
        current_items = input()
        if current_items == 'Finish': break
        items_count += 1
        current_sum += price_dict[current_items]


    if not items_count & 1:
        current_sum *= 0.80
    total_sum += current_sum
    print(f'You purchased {items_count} items for {current_sum:.2f} leva.')


print(f'Average bill per client is: {total_sum / number_clients:.2f} leva.')


'''
1
basket
wreath
chocolate bunny
wreath
basket
chocolate bunny
Finish
'''

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 20 and 21 April 2019/06. Easter Decoration/06. Easter Decoration.js
function easterDecoration(arg) {
    let numberClients = arg.shift()
    let subtotalSum = 0
    let totalSum = 0
    const priceObj = { 'basket': 1.50, 'wreath' : 3.80, 'chocolate bunny' : 7}

    for (let i = 0; i < numberClients; i++) {
        subtotalSum = 0
        let itemCount = 0
        while (true) {
            current = arg.shift()
            if (current !== 'Finish') {
                subtotalSum += priceObj[current]
                itemCount++
            } else {
                itemCount % 2 === 0 ? subtotalSum *= 0.80 : null
                totalSum += subtotalSum
                console.log(`You purchased ${itemCount} items for ${subtotalSum.toFixed(2)} leva.`)
                break
            }
        }
    }

    console.log(`Average bill per client is: ${(totalSum / numberClients).toFixed(2)} leva.`)
}

//easterDecoration([2, 'basket', 'wreath', 'chocolate bunny', 'Finish', 'wreath', 'chocolate bunny', 'Finish'])
//easterDecoration([1, 'basket','wreath','chocolate bunny','wreath','basket','chocolate bunny','Finish'])